From 0dee91880c1cdb78a42ec0f4710c7457442ec34b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=A9r=C3=B4me=20Guelfucci?= <jeromeg@xfce.org>
Date: Fri, 21 May 2010 10:14:26 +0200
Subject: [PATCH] Fix a crash when the config file does not exist yet (bug
 #6460).

Calling xfce_rc_close on a NULL rc file is not exactly a good idea.
Patch by Thomas (tg42@gmx.de).
---
 lib/screenshooter-utils.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/lib/screenshooter-utils.c b/lib/screenshooter-utils.c
index 9615c66..ec84657 100644
--- a/lib/screenshooter-utils.c
+++ b/lib/screenshooter-utils.c
@@ -94,11 +94,11 @@ screenshooter_read_rc_file (const gchar *file, ScreenshotData *sd)
           g_free (title);
           title =
             g_strdup (xfce_rc_read_entry (rc, "title", _("Screenshot")));
-        }
 
-      TRACE ("Close the rc file");
+          TRACE ("Close the rc file");
 
-      xfce_rc_close (rc);
+          xfce_rc_close (rc);
+        }
     }
 
   /* And set the sd values */
-- 
1.7.9

